src/lua/utils.c

```diff
@@ -328,7 +342,43 @@ luaL_register_module(struct lua_State *L, const char *modname,
 			luaL_error(L, "Failed to register library");
 	}
 	lua_remove(L, -2);  /* remove _LOADED table */
-	luaL_register(L, NULL, methods);
+	luaT_setfuncs(L, methods);
+}
+
+int
+luaT_newmodule(struct lua_State *L, const char *modname,
+	       const struct luaL_Reg *funcs)
+{
+	say_debug("%s(%s)", __func__, modname);
+	assert(modname != NULL && funcs != NULL);
+
+	/* Get package.loaded. */
+	lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
+
+	/* Verify that the module is not already registered. */
+	lua_getfield(L, -1, modname);
+	assert(lua_isnil(L, -1));
+	lua_pop(L, 1);
+
+	/* Create a module table. */
+	lua_newtable(L);
```

Comment:
Side note: I wondering: can we use `lua_createtable()` instead (as far as we can count necessary array length). But `luaT_newmodule()` doesn't look like performance critical function, so it's OK to use `lua_newtable()` here.

Reply:
At cost of caller's code complication (either by adding the argument or by adding the argument and wrapping it into a macro to hide the complication). For my taste, it is better to keep it simple.

Reply:
Yes, its totally fine for me.